brew-rmtree.rb

```diff
@@ -29,61 +29,67 @@ def bash(command)
     return %x! bash -c #{escaped_command} !
   end
 
-  def rmtree
-    if ARGV.size < 1 or ['-h', '?', '--help'].include? ARGV.first
-      puts USAGE
-      exit 0
-    end
-
-    raise KegUnspecifiedError if ARGV.named.empty?
-
-    if not ARGV.force?
-      ARGV.named.each do |keg_name|
+  def remove_keg(keg_name)
+    # Remove old versions of keg
+    puts bash "brew cleanup #{keg_name}"
 
-        # Remove old versions of keg
-        puts bash "brew cleanup #{keg_name}"
-
-        # Remove current keg
-        puts bash "brew uninstall #{keg_name}"
-
-        deps = bash "join <(brew leaves) <(brew deps #{keg_name})"
-
-        if deps.length > 0
-          puts "Found lingering dependencies"
-          puts deps.chomp
+    # Remove current keg
+    puts bash "brew uninstall #{keg_name}"
+  end
 
-          deps = deps.split("\n")
+  def deps(keg_name)
+    deps = bash "join <(brew leaves) <(brew deps #{keg_name})"
+    deps.split("\n")
+  end
 
-          deps.each do |dep|
-            dep = dep.chomp
+  def reverse_deps(keg_name)
+    reverse_deps = bash "brew uses --installed #{keg_name}"
+    reverse_deps.split("\n")
+  end
 
-            if !dep.empty?
-              # Check if anything currently installed uses the dependency
-              dep_deps = bash "brew uses --installed #{dep}"
-              dep_deps = dep_deps.chomp
+  def rmtree(keg_name)
+    # Check if anything currently installed uses the keg
+    reverse_deps = reverse_deps(keg_name)
 
-              if dep_deps.length > 0
-                puts "Not removing dependency #{dep} because other installed packages depend on it:"
-                puts dep_deps
-              else
-                # Nothing claims to depend on it
-                puts "Removing dependency #{dep}..."
-                puts bash "brew rmtree #{dep}"
-              end
-            end
-          end
-        else
-          puts "No dependencies left on system for #{keg_name}."
+    if reverse_deps.length > 0
+      puts "Not removing #{keg_name} because other installed kegs depend on it:"
```

Comment:
Yes, sorry, I should have made my intentions clearer. I deliberately made it so that the kegs named in the original command will not be uninstalled if there are other kegs that depend on them. This was the behavior I expected when installing, and I was surprised when the tool happily allowed me to uninstall `python` despite quite a few Formulae depending on it.

Of course, if you disagree with this change, then by all means do away with it. I can just tap my own fork for future use. But I do agree that it is "safer", and I doubt I am the only user who initially expected this behavior.

Perhaps the original behavior could be preserved by passing the `--force` option?